client/src/main/java/org/eclipse/hono/client/impl/AbstractHonoClient.java

```diff
@@ -70,6 +70,8 @@
      */
     protected List<Symbol> offeredCapabilities = Collections.emptyList();
 
+    private long autoCloseLinksTimerId = -1;
```

Comment:
I think it would be better using a `Long` here and initializing the value with `null`. Then there can be a null check added around the cancelTimer invocation and, as first thing inside the timer handler in `startAutoCloseLinksTimer`, the `autoCloseLinksTimerId` can be reset to `null`.

Reply:
> Then there can be a null check added around the cancelTimer invocation 

The reason to use `long` instead of `Long`, that I do not _need_ the `null` check :smile: (`cancelTimer` does nothing for a negative ID). But yes, I see that `null` is more obvious.